### PR TITLE
Refactor `iris.experimental.ugrid`

### PR DESCRIPTION
**Pull request by trexfeathers**
### Description

Splitting the module into submodules. Looking forward to the conflicts!

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: lib/iris/experimental/ugrid/__init__.py [hunk 1]
```diff
@@ -8,4029 +8,29 @@
...
-from ...fileformats._nc_load_rules.helpers import get_attr_units, get_names
-from ...io import decode_uri, expand_filespecs
-from ...util import guess_coord_axis
 
 __all__ = [
     "Connectivity",
-    "ConnectivityMetadata",
     "load_mesh",
     "load_meshes",
     "save_mesh",
```
Comment (bjlittle) [inline lib/iris/experimental/ugrid/__init__.py:32]: @trexfeathers You've inherited this... but I *think* we're sorting alphabetical (exclusive of case), so `save_mesh` should be last, right?

File: lib/iris/experimental/ugrid/__init__.py [hunk 2]
```diff
@@ -8,4029 +8,29 @@
...
-from collections import namedtuple
-from collections.abc import Iterable
-from contextlib import contextmanager
-from functools import wraps
-from itertools import groupby
-import logging
-from pathlib import Path
-import re
-import threading
+logger = get_logger(__name__, fmt="[%(cls)s.%(funcName)s]")
```
Comment (bjlittle) [inline lib/iris/experimental/ugrid/__init__.py:20]: @trexfeathers This isn't quite right... but I'm tempted to bank this as is, and we fix it in a separate PR.

**Review by bjlittle — CHANGES_REQUESTED** [review]
@trexfeathers Thanks for taking this refactor on... it's been well overdue :+1: 

Just a minor comment to service about `__all__` sort order, and the logging we can fix in a follow-up PR; this PR is noisy enough.

Also, I think this is worthy of a `whatsnew` entry... fancy adding one? :wink:

File: lib/iris/experimental/ugrid/__init__.py (hunk 1 above)
Comment (trexfeathers) [reply lib/iris/experimental/ugrid/__init__.py]: c71b88751da2d0c73ebf482ddc4e9ef6127a1934

**Review by bjlittle — APPROVED** [review]

**Pull request merged**